scripts/run-plugin-nuts.js

```diff
@@ -0,0 +1,201 @@
+#!/usr/bin/env node
+const path = require('path');
+const fs = require('fs');
+const execSync = require('child_process').execSync;
+const got = require('got');
+const { isArray } = require('@salesforce/ts-types');
+
+const circleciBaseUrl = 'https://circleci.com/api/v2/';
+
+/**
+ * Script to locate and launch NUTs across sfdx modules
+ *
+ * Script requires that the env var CIRCLECI_API_TOKEN is set to a valid CircleCi API Token
+ *
+ * args:
+ *   sfdx-cli version: z.y.z | latest | latest-rc
+ *
+ * Flow:
+ *   load sfdx-cli package.json
+ *   For each entry in oclif.plugins
+ *     Identify repo organization and project i.e. @salesforce/plugin-auth -> salesforcecli/plugin-auth via "npm show @salesforce/plugin-auth homepage"
+ *     Identify if plugin supports script test:nuts i.e. "npm show @salesforce/plugin-auth scripts | grep test:nuts"
+ *     Check if org/project is known in circleci
+ *       Trigger test nuts workflow for plugin with required parameters
+ */
+
+const manualRepoOverrides = {
+  'salesforce-alm': {
+    org: 'salesforce-cli',
+    repo: 'toolbelt',
+  },
+};
+
+const hasTestNuts = (module) => {
+  const results = execSync(`npm show ${module.name}@${module.version} scripts`);
+  return results.includes('test:nuts');
+};
+
+const getNpmForModule = (timeCreated, module, sections) => {
+  const results = execSync(
+    `npm show ${module.name}@${module.version} ${['time', 'version', ...sections].join(' ')} --json`
+  );
+  let jsonResults = JSON.parse(results);
+  if (isArray(jsonResults)) {
+    jsonResults = jsonResults
+      .sort((a, b) => new Date(b.time[b.version]).getTime() - new Date(a.time[a.version]).getTime())
+      .find((result) => new Date(result.time[result.version]).getTime() <= timeCreated.getTime());
+  }
+  return jsonResults;
+};
+
+const getOrgAndProject = (module) => {
+  const results = execSync(`npm show ${module.name} homepage`);
+  if (Reflect.ownKeys(manualRepoOverrides).includes(module.name)) {
+    return manualRepoOverrides[module.name];
+  }
+  const repoUrl = new URL(results.toString().trim());
+  const [, org, repo] = repoUrl.pathname.split('/');
+  return { org, repo };
+};
+
+const orgSlug = (org, repo) => {
+  return `github/${org}`;
+};
+const projectSlug = (org, repo) => {
+  return `${orgSlug(org)}/${repo}`;
+};
+
+/**
+ * Triggers the just-nuts workflow for the named org/repo/branch
+ * @param sfdxVersion
+ * @param branch
+ * @returns {Promise<*>}
+ */
+const triggerNutsForProject = async (sfdxVersion, module, branch = 'main') => {
+  const body = {
+    branch: branch,
+    parameters: {
+      'run-auto-workflows': false,
+      'run-just-nuts': true,
+      sfdx_version: sfdxVersion,
+      repo_tag: module.version,
```

Comment:
Passes the module.version found for a plugin as the repo_tag that will be used in checkout.